.github/workflows/code-quality.yml

```diff
@@ -48,5 +48,5 @@ jobs:
           python -m pip install black
 
       - name: Check Python formatting
-        run: black --check .
+        run: black --diff --check .
```

Comment:
I'd rather pin and have a process to upgrade our pinned versions regularly (just do it if trivial, make issues if not).  

If that's not practical, then I'd go with latest and greatest, with an option to pin if desired during a PR rather than pollute with formatting changes... which is basically compatible with what we're already doing.

For background, what prompted this specific change was that, despite having the same version of black and same OS as CI, my local black scan didn't pick up the issue (a couple of extraneous newlines).  I didn't look into it too deeply.  I initially assumed it was an artefact of jumping between Windows and Linux (i.e. line endings), but I don't think that holds up to scrutiny.

Reply:
Ah, should have read more comments, i.e. https://github.com/OpenAssetIO/OpenAssetIO-TraitGen/pull/12#discussion_r1133598615

So perhaps I had/have some config issue with black version :thinking:

Reply:
Sounds good re pinning  - would you mind making an issue?